bluesky/run_engine.py

```diff
@@ -464,9 +464,9 @@ def __call__(self, plan, subs=None, **metadata):
 
         # Register temporary subscriptions. Save tokens to unsubscribe later.
         subs = normalize_subs_input(subs)
-        try:
+        if hasattr(plan, 'subs'):
```

Comment:
...But if this is annoying to you I'll reverse it in both cases. Readability should be the priority here.